src/Simplify.elm

```diff
@@ -3278,6 +3305,105 @@ consChecks checkInfo =
+                    if (earlierFn == inverseFn0 && laterFn == inverseFn1) || (earlierFn == inverseFn1 && laterFn == inverseFn0) then
+                        Just
+                            (Rule.errorWithFix
+                                { message = qualifiedToString earlierFn ++ " and " ++ qualifiedToString laterFn ++ " cancel each other out"
+                                , details = config.details
+                                }
+                                config.laterFnInfo.range
+                                config.fix
+                            )
+
+                    else
+                        Nothing
+
+                _ ->
+                    Nothing
+    in
+    firstThatConstructsJust
+        [ \() ->
+            case ( AstHelpers.getValueOrFunction checkInfo.earlier, AstHelpers.getComposition checkInfo.later ) of
+                ( Just earlierFnInfo, Just composition ) ->
+                    case AstHelpers.getValueOrFunction composition.earlier of
+                        Just laterFnInfo ->
+                            checkFnInfosForInverses
+                                { earlierFnInfo = earlierFnInfo
+                                , laterFnInfo = laterFnInfo
+                                , details = [ "You can remove these two functions." ]
+                                , fix =
+                                    keepOnlyFix { parentRange = checkInfo.parentRange, keep = Node.range checkInfo.later }
+                                        ++ removeAndBetweenFix { remove = Node.range composition.earlier, keep = Node.range composition.later }
+                                }
+
+                        Nothing ->
+                            Nothing
+
+                _ ->
+                    Nothing
+        , \() ->
+            case ( AstHelpers.getComposition checkInfo.earlier, AstHelpers.getValueOrFunction checkInfo.later ) of
+                ( Just composition, Just laterFnInfo ) ->
+                    case AstHelpers.getValueOrFunction composition.later of
+                        Just earlierFnInfo ->
+                            checkFnInfosForInverses
+                                { earlierFnInfo = earlierFnInfo
+                                , laterFnInfo = laterFnInfo
+                                , details = [ "You can remove these two functions." ]
+                                , fix =
+                                    keepOnlyFix { parentRange = checkInfo.parentRange, keep = Node.range checkInfo.earlier }
+                                        ++ removeAndBetweenFix { remove = Node.range composition.later, keep = Node.range composition.earlier }
+                                }
+
+                        Nothing ->
+                            Nothing
+
+                _ ->
+                    Nothing
+        , \() ->
+            case ( AstHelpers.getValueOrFunction checkInfo.earlier, AstHelpers.getValueOrFunction checkInfo.later ) of
```

Comment:
Should we extract these `getCompsoition` and `getValueOrFunction` calls to variables to avoid computing them multiple times?

Reply:
fixed.